.github/workflows/push-pull.yml

```diff
@@ -73,26 +77,40 @@ jobs:
           # To get around this, always make the key unique and rely on
           # restore-keys to find the highest-indexed key available. Explicitly
           # try to restore things from the most recent builds.
-          key: ${{ runner.os }}-build-${{ github.run_number }}
+          key: ${{ runner.os }}-build-cmake-${{ github.run_number }}
           restore-keys: |
-            ${{ runner.os }}-build-${{ steps.keys.outputs.key1 }}
-            ${{ runner.os }}-build-${{ steps.keys.outputs.key2 }}
-            ${{ runner.os }}-build-${{ steps.keys.outputs.key3 }}
-            ${{ runner.os }}-build-${{ steps.keys.outputs.key4 }}
-            ${{ runner.os }}-build-${{ steps.keys.outputs.key5 }}
-            ${{ runner.os }}-build-
+            ${{ runner.os }}-build-cmake-${{ steps.keys.outputs.key1 }}
+            ${{ runner.os }}-build-cmake-${{ steps.keys.outputs.key2 }}
+            ${{ runner.os }}-build-cmake-${{ steps.keys.outputs.key3 }}
+            ${{ runner.os }}-build-cmake-${{ steps.keys.outputs.key4 }}
+            ${{ runner.os }}-build-cmake-${{ steps.keys.outputs.key5 }}
+            ${{ runner.os }}-build-cmake-${{ steps.keys.outputs.key10 }}
```

Comment:
Why does this jump from 5 to 10?

Reply:
Originally I thought it might help to provide a specific key to look for in case the last five don't yet exist (like if we do some combination of creating PRs and merging PRs that starts five or more jobs at once) but in that case GitHub should default to just picking the next newest key (though this isn't documented; I'm guessing how they decide which key is best). I don't have any evidence that picking 'ten keys ago' will work any better than leaving GitHub to its own choices so I'll delete that line.